MetaBAT installation failure because libhts.a required -lcurl (which requires -lssl)

The installation of `MetaBAT-2.12.1-foss-2018b-Python-2.7.15.eb` got broken due to adding `cURL` as dependency to `SAMtools` (cfr. #8131), since `libhts.a` (provided by `SAMtools`) now required also linking with `-lcurl`, which in turns requires additional libraries to be linked in (`-lopenssl` & co).

Problem manifests as `undefined reference` error for `samtools/lib/libhts.a`:

```
samtools/lib/libhts.a(hfile_libcurl.o):hfile_libcurl.c:function easy_errno: error: undefined reference to 'curl_easy_getinfo'
samtools/lib/libhts.a(hfile_libcurl.o):hfile_libcurl.c:function easy_errno: error: undefined reference to 'curl_easy_getinfo'
samtools/lib/libhts.a(hfile_libcurl.o):hfile_libcurl.c:function wait_perform: error: undefined reference to 'curl_multi_fdset'
```

It's not a simple as just adding `curl` to the list of libraries included in the patch file, since:

```
$ pkg-config --libs --static libcurl
-L/user/gent/400/vsc40023/eb_scratch/CO7/haswell-ib/software/cURL/7.60.0-GCCcore-7.3.0/lib -lcurl -lssl -lcrypto -lssl -lcrypto -lz -lm -lpthread
```

@SethosII Any idea if we can do a better job with `libhts.a`, by "baking in" the libraries it requires (and moving `cURL` and maybe more to `builddependencies` for SAMtools)?

--- a/easybuild/easyconfigs/m/MetaBAT/MetaBAT-2.12.1-foss-2018b-Python-2.7.15.eb
+++ b/easybuild/easyconfigs/m/MetaBAT/MetaBAT-2.12.1-foss-2018b-Python-2.7.15.eb
@@ -14,10 +14,13 @@ sources = ['v%(version)s.tar.gz']
 patches = ['MetaBAT-%(version)s_fix-prepend-PATH.patch']
 checksums = [
     'e3aca0656f56f815135521360dc56667ec26af25143c3a31d645fef1a96abbc2',  # v2.12.1.tar.gz
-    '2e5f84ad7090e7853c506278adf2d09a8af57b90d7003c5c01a62fa58c9f4931',  # MetaBAT-2.12.1_fix-prepend-PATH.patch
+    '7373c11ba9b86dddbfe5a472448b7fb6589d3ff4f4882a9829d14f47aec32d8f',  # MetaBAT-2.12.1_fix-prepend-PATH.patch
 ]
 
-builddependencies = [('SCons', '3.0.4', versionsuffix)]
+builddependencies = [
+    ('pkg-config', '0.29.2'),
+    ('SCons', '3.0.4', versionsuffix),
+]
 
 dependencies = [
     ('Python', '2.7.15'),

--- a/easybuild/easyconfigs/m/MetaBAT/MetaBAT-2.12.1_fix-prepend-PATH.patch
+++ b/easybuild/easyconfigs/m/MetaBAT/MetaBAT-2.12.1_fix-prepend-PATH.patch
@@ -1,9 +1,11 @@
 prepend to $PATH rather than append, to avoid picking up system compiler (g++);
-add missing -llzma -lbz2 link options, required when linking to libhts.a
-author: Kenneth Hoste (HPC-UGent)
 see also https://bitbucket.org/berkeleylab/metabat/issues/52
+
+also disable static linking and add missing link options required by libhts.a via pkg-config (-llzma -lbz2 -lcurl & co)
+
+author: Kenneth Hoste (HPC-UGent)
 --- berkeleylab-metabat-37db58fe3fda/SConstruct.orig	2017-09-01 07:28:02.000000000 +0200
-+++ berkeleylab-metabat-37db58fe3fda/SConstruct	2019-03-21 12:24:35.516202521 +0100
++++ berkeleylab-metabat-37db58fe3fda/SConstruct	2019-11-26 15:22:10.881115082 +0100
 @@ -7,7 +7,7 @@
  def ENV_update(tgt_ENV, src_ENV):
      for K in src_ENV.keys():
@@ -12,22 +14,41 @@ see also https://bitbucket.org/berkeleylab/metabat/issues/52
 +            tgt_ENV[K]=SCons.Util.PrependPath(tgt_ENV[K], src_ENV[K])
          else:
              tgt_ENV[K]=src_ENV[K]
- 
-@@ -184,7 +184,7 @@
+
+@@ -27,7 +27,7 @@
+ link_flags = ['-lstdc++', '-lm', '-fopenmp']
+
+ if platform.platform(True, True).find('Darwin') == -1:
+-    link_flags.extend(['-static', '-static-libgcc', '-static-libstdc++'])
++    link_flags.extend(['-static-libgcc', '-static-libstdc++'])
+
+ if debug is None:
+     build_flags.extend(['-O3', '-DNDEBUG', '-Wno-unknown-pragmas', '-Wno-deprecated-declarations', '-Wno-overflow', '-Wno-unused-variable'])
+@@ -178,13 +178,17 @@
+ if not isOldSamtools:
+     findStaticOrShared('hts', hts_lib, sources, linkflags)
+
++env.ParseConfig("pkg-config --libs --static liblzma")
++env.ParseConfig("pkg-config --libs --static bzip2")
++env.ParseConfig("pkg-config --libs --static libcurl")
++
+ ccflags = build_flags
+ ccflags.append(hts_flag)
+ jgi_summarize_bam_contig_depths = env.Program("jgi_summarize_bam_contig_depths",
                  sources,
                  CCFLAGS=ccflags,
                  CPPPATH=[hts_inc, boost_inc],
 -                LIBS=['pthread', 'z'],
-+                LIBS=['pthread', 'z', 'lzma', 'bz2'],
++                #LIBS=['pthread', 'z'],
                  LINKFLAGS=linkflags
                  )
- 
-@@ -204,7 +204,7 @@
+
+@@ -204,7 +208,7 @@
                  CPPPATH=[hts_inc, boost_inc],
                  LINKPATH=[hts_lib, boost_lib],
                  LINKFLAGS=linkflags,
 -                LIBS=['z']
-+                LIBS=['z', 'lzma', 'bz2']
++                #LIBS=['z'],
                  )
- 
+
  test_sum = env.Alias('test_sum', [jgi_summarize_bam_contig_depths],  ' '.join([jgi_summarize_bam_contig_depths[0].abspath, '--outputDepth', test_depth, test_bam]) )

